graql/reasoner/query/ReasonerQueryImpl.java

```diff
@@ -169,15 +171,31 @@ public CompositeQuery asComposite() {
         return new CompositeQuery(getPattern(), traversalExecutor, context());
```

Comment:
We should try to make this throw as we shouldn't be able to have a CompositeQuery without negations. Keep in mind that this class should then become `ConjunctiveQuery`

Reply:
To be addressed in a separate PR as a refactor